### PR TITLE
Add credential and credential types endpoint

### PR DESCRIPTION
**Pull request by lpichler**
TPINVTRY-726
Api endpoints for service credential and service  credential types
and  credentials subcollections for service_offerings,service_instances,. ..

### CODE REVIEW

**Comment by agrare** [comment]
@lpichler can you bump the openapi version so we can do a point release of the client?

**Comment by lpichler** [comment]
> @lpichler can you bump the openapi version so we can do a point release of the client?

Done

**Comment by lpichler** [comment]
I also added `ServiceCredentials` subcollections for `ServinceInstance, ServinceOffering, ServinceInstanceNode, ServinceInstanceNode`

and I did it like we have It for tagging, I used one controller(`Credentialings`) 
but `naming is hard` (I heard this quote from JoeR :) ). If you have better idea let me know. 

thanks!


cc @bdunne

File: app/controllers/api/v2x0/service_credential_types_controller.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module Api
+  module V2x0
```
Comment (bdunne) [inline app/controllers/api/v2x0/service_credential_types_controller.rb]: Since we're following simver and v2.0 is already public, this should be in v2.1

**Comment by gtanzillo** [comment]
I agree with Brandon's concerns...

I was not sure that the "tagging" pattern fits this use case and wanted to hear what others thought. I think it detracts more than it adds and it makes more sense to keep it simple.

Also, we'll need to have some tests here, as Brandon suggested.

**Comment by slemrmartin** [comment]
@bdunne relations are similar to tags. We have 4 models, 4 join tables and 1 credentials (so 4x `has many, :through`) - tags have the same structure.

I would rather see paths like `service_offerings/:id/service_credentials` etc., but it's not normal has_many, so I don't know how it works with all the mixins

**Comment by lpichler** [comment]
> * Does this need to be like tags with a special controller?  I'd prefer it to act more like a normal `has_many` relation if possible.

@bdunne  I guess you started with this ? any progress about it ? thanks!

File: app/controllers/api/v2x0/service_credential_types_controller.rb (hunk 1 above)
Comment (bdunne) [reply app/controllers/api/v2x0/service_credential_types_controller.rb]: Do you mind basing this on v3.0 from https://github.com/RedHatInsights/topological_inventory-api/pull/291 ?

**Comment by bdunne** [comment]
> > * Does this need to be like tags with a special controller?  I'd prefer it to act more like a normal `has_many` relation if possible.
> 
> @bdunne I guess you started with this ? any progress about it ? thanks!

Yes, this was merged in https://github.com/RedHatInsights/topological_inventory-api/pull/288

**Comment by lpichler** [comment]
> > @bdunne I guess you started with this ? any progress about it ? thanks!
> 
> Yes, this was merged in #288

yes it works, 

PR updated 👍 

thanks

**Comment by lpichler** [comment]
Client PR https://github.com/RedHatInsights/topological_inventory-api-client-ruby/pull/29

**Pull request merged**